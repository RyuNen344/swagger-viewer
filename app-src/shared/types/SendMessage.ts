export type MessageType = string

export type ExecConvertSwaggerMessage = {
  type: MessageType
}
