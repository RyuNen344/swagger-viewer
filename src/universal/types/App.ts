export const APP_RENDER_ID = "swagger-viewer--render-root"

export const ORIGINAL_SRC_AREA_CLASS = "swagger-viewer--original_src_area"

export type MessagePayload = {
  type: "CALL_CONVERT_SWAGGER"
}
