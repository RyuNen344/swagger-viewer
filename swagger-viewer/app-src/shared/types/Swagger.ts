import { CastAny } from "./utils"

export type OpenApi2Json = Readonly<{
  info: Record<string, CastAny>
  openapi: undefined
  paths: Record<string, CastAny>
  swagger: "2.0"
}>

export type OpenApi3Json = Readonly<{
  info: Record<string, CastAny>
  openapi: "3.0" | "3.0.0" | string
  paths: Record<string, CastAny>
  servers: ReadonlyArray<Record<string, CastAny>>
  swagger: undefined
}>

export type NotOpenApiJson = Readonly<{
  openapi: undefined
  swagger: undefined
}>

export type MaybeSwaggerJson = OpenApi2Json | OpenApi3Json | NotOpenApiJson
