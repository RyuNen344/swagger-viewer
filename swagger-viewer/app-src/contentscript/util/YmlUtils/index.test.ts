import { CastAny } from "../../../shared/types/utils"
import { yml as case2_yml } from "./fixtures/case2"
import { convertToObject } from "."

describe("convertToObject / convertable", () => {
  it("null", () => {
    // ## Arrange ##
    const src = null
    // ## Act ##
    const result = convertToObject(src as CastAny)
    // ## Assert ##
    expect(result).toMatchInlineSnapshot(`null`)
  })

  it("empty string", () => {
    // ## Arrange ##
    const src = ``
    // ## Act ##
    const result = convertToObject(src)
    // ## Assert ##
    expect(result).toMatchInlineSnapshot(`null`)
  })

  it("no swagger string", () => {
    // ## Arrange ##
    const src = `any string literal`
    // ## Act ##
    const result = convertToObject(src)
    // ## Assert ##
    expect(result).toMatchInlineSnapshot(`null`)
  })

  it("yaml string", () => {
    // ## Arrange ##
    const src = `
swagger: "2.0"
info:
  title: Simple API overview
paths:
  /:
    get:
      operationId: listVersionsv2
      produces:
      - application/json
      responses:
        "200":
          description: |-
            200 300 response
consumes:
- application/json
`
    // ## Act ##
    const result = convertToObject(src)
    // ## Assert ##
    expect(result).toMatchInlineSnapshot(`
      Object {
        "consumes": Array [
          "application/json",
        ],
        "info": Object {
          "title": "Simple API overview",
        },
        "paths": Object {
          "/": Object {
            "get": Object {
              "operationId": "listVersionsv2",
              "produces": Array [
                "application/json",
              ],
              "responses": Object {
                "200": Object {
                  "description": "200 300 response",
                },
              },
            },
          },
        },
        "swagger": "2.0",
      }
    `)
  })

  it("yaml string case2", () => {
    // ## Arrange ##
    // ## Act ##
    const result = convertToObject(case2_yml)
    // ## Assert ##
    expect(result).toMatchSnapshot()
  })

  it("json string", () => {
    // ## Arrange ##
    const src = `
{
  "a": "a",
  "b": 2,
  "cc": null,
}
`
    // ## Act ##
    const result = convertToObject(src)
    // ## Assert ##
    expect(result).toMatchInlineSnapshot(`
      Object {
        "a": "a",
        "b": 2,
        "cc": null,
      }
    `)
  })
})

describe("convertToObject / broken string", () => {
  describe("broken json string", () => {
    it("case:1", () => {
      // ## Arrange ##
      const src = `
{
  "a": "a",
  "b": 2,
  "cc": null,
`
      // ## Act ##
      // ## Assert ##
      expect(() => {
        convertToObject(src)
      }).toThrowErrorMatchingInlineSnapshot(`
"unexpected end of the stream within a flow collection at line 6, column 1:
    
    ^"
`)
    })

    it("case:2", () => {
      // ## Arrange ##
      const src = `
{
  "a": "a"
  "b": 2,
  "cc": null,
}
`
      // ## Act ##
      // ## Assert ##
      expect(() => {
        convertToObject(src)
      }).toThrowErrorMatchingInlineSnapshot(`
"missed comma between flow collection entries at line 4, column 3:
      \\"b\\": 2,
      ^"
`)
    })
  })

  describe("broken yaml string", () => {
    it("case:1", () => {
      // ## Arrange ##
      const src = `
swagger: "2.0"
info:
                title: Simple API overview
paths:
  /:
    get: operationId: listVersionsv2
      produces:
      - application/json
      responses:
        "200":
          description: |-
            200 300 response
consumes:
- application/json
`
      // ## Act ##
      // ## Assert ##
      expect(() => {
        convertToObject(src)
      }).toThrowErrorMatchingInlineSnapshot(`
"incomplete explicit mapping pair; a key node is missed; or followed by a non-tabulated empty line at line 7, column 21:
        get: operationId: listVersionsv2
                        ^"
`)
    })

    it("case:2", () => {
      // ## Arrange ##
      const src = `
swagger: "2.0"
info:
  title: Simple API overview
paths:
  /:
    get:
              operationId: listVersionsv2
      produces:
application/json
      responses:
        "200":
          description: |-
            200 300 response
consumes:

`
      // ## Act ##
      // ## Assert ##
      expect(() => {
        convertToObject(src)
      }).toThrowErrorMatchingInlineSnapshot(`
"bad indentation of a mapping entry at line 9, column 7:
          produces:
          ^"
`)
    })
  })
})
