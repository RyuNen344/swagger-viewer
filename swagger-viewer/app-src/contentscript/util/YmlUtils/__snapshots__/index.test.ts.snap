// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToObject yaml string case2 1`] = `
Object {
  "info": Object {
    "title": "Simple API overview",
    "version": "2.0.0",
  },
  "openapi": "3.0.0",
  "paths": Object {
    "/": Object {
      "get": Object {
        "operationId": "listVersionsv2",
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "examples": Object {
                  "foo": Object {
                    "value": Object {
                      "versions": Array [
                        Object {
                          "id": "v2.0",
                          "links": Array [
                            Object {
                              "href": "http://127.0.0.1:8774/v2/",
                              "rel": "self",
                            },
                          ],
                          "status": "CURRENT",
                          "updated": "2011-01-21T11:33:21Z",
                        },
                        Object {
                          "id": "v3.0",
                          "links": Array [
                            Object {
                              "href": "http://127.0.0.1:8774/v3/",
                              "rel": "self",
                            },
                          ],
                          "status": "EXPERIMENTAL",
                          "updated": "2013-07-23T11:33:21Z",
                        },
                      ],
                    },
                  },
                },
              },
            },
            "description": "200 response",
          },
          "300": Object {
            "content": Object {
              "application/json": Object {
                "examples": Object {
                  "foo": Object {
                    "value": "{
 \\"versions\\": [
       {
         \\"status\\": \\"CURRENT\\",
         \\"updated\\": \\"2011-01-21T11:33:21Z\\",
         \\"id\\": \\"v2.0\\",
         \\"links\\": [
             {
                 \\"href\\": \\"http://127.0.0.1:8774/v2/\\",
                 \\"rel\\": \\"self\\"
             }
         ]
     },
     {
         \\"status\\": \\"EXPERIMENTAL\\",
         \\"updated\\": \\"2013-07-23T11:33:21Z\\",
         \\"id\\": \\"v3.0\\",
         \\"links\\": [
             {
                 \\"href\\": \\"http://127.0.0.1:8774/v3/\\",
                 \\"rel\\": \\"self\\"
             }
         ]
     }
 ]
}
",
                  },
                },
              },
            },
            "description": "300 response",
          },
        },
        "summary": "List API versions",
      },
    },
    "/v2": Object {
      "get": Object {
        "operationId": "getVersionDetailsv2",
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "examples": Object {
                  "foo": Object {
                    "value": Object {
                      "version": Object {
                        "id": "v2.0",
                        "links": Array [
                          Object {
                            "href": "http://127.0.0.1:8774/v2/",
                            "rel": "self",
                          },
                          Object {
                            "href": "http://docs.openstack.org/api/openstack-compute/2/os-compute-devguide-2.pdf",
                            "rel": "describedby",
                            "type": "application/pdf",
                          },
                          Object {
                            "href": "http://docs.openstack.org/api/openstack-compute/2/wadl/os-compute-2.wadl",
                            "rel": "describedby",
                            "type": "application/vnd.sun.wadl+xml",
                          },
                          Object {
                            "href": "http://docs.openstack.org/api/openstack-compute/2/wadl/os-compute-2.wadl",
                            "rel": "describedby",
                            "type": "application/vnd.sun.wadl+xml",
                          },
                        ],
                        "media-types": Array [
                          Object {
                            "base": "application/xml",
                            "type": "application/vnd.openstack.compute+xml;version=2",
                          },
                          Object {
                            "base": "application/json",
                            "type": "application/vnd.openstack.compute+json;version=2",
                          },
                        ],
                        "status": "CURRENT",
                        "updated": "2011-01-21T11:33:21Z",
                      },
                    },
                  },
                },
              },
            },
            "description": "200 response",
          },
          "203": Object {
            "content": Object {
              "application/json": Object {
                "examples": Object {
                  "foo": Object {
                    "value": Object {
                      "version": Object {
                        "id": "v2.0",
                        "links": Array [
                          Object {
                            "href": "http://23.253.228.211:8774/v2/",
                            "rel": "self",
                          },
                          Object {
                            "href": "http://docs.openstack.org/api/openstack-compute/2/os-compute-devguide-2.pdf",
                            "rel": "describedby",
                            "type": "application/pdf",
                          },
                          Object {
                            "href": "http://docs.openstack.org/api/openstack-compute/2/wadl/os-compute-2.wadl",
                            "rel": "describedby",
                            "type": "application/vnd.sun.wadl+xml",
                          },
                        ],
                        "media-types": Array [
                          Object {
                            "base": "application/xml",
                            "type": "application/vnd.openstack.compute+xml;version=2",
                          },
                          Object {
                            "base": "application/json",
                            "type": "application/vnd.openstack.compute+json;version=2",
                          },
                        ],
                        "status": "CURRENT",
                        "updated": "2011-01-21T11:33:21Z",
                      },
                    },
                  },
                },
              },
            },
            "description": "203 response",
          },
        },
        "summary": "Show API version details",
      },
    },
  },
}
`;
